src/libraries/Native/Unix/System.Native/pal_networkchange.c

```diff
@@ -162,4 +164,12 @@ void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)
         }
     }
 }
+#else
+void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)
+{
+    // NOTSUPPORTEDAPI
```

Comment:
`NOTSUPPORTEDAPI` was just a marker for myself to find all these at the end - in case if I get a better idea what should be in the bodies. I forgot to remove these. 

These are methods that were previously excluded on some platforms via conditional compilation due to underlying OS not having said functionality. The stubs that I have put instead are primarily to make compiler happy. They are not expected to be called, since previously they were not even there. 
In most cases there is a way to fail gracefully with ENOSUP, which I think is the right way to implement these. However a few cases do not provide such option.

For the cases that cannot naturally fail, my preference would be to have a "release assert" to prevent accidental use of unimplemented API. `__builtin_unreachable` is one way to achieve that which is also cross-platform.

Reply:
I could also use `abort()`. I will switch to that as a more obvious failfast case.

Reply:
>I could also use abort(). I will switch to that as a more obvious failfast case.

That would be my preference here. Thanks.

Reply:
> In most cases there is a way to fail gracefully with ENOSUP, which I think is the right way to implement these.

There is a quite a bit of prior art in the libraries PAL where we handle this case by returning `errno = ENOTSUP`. I do not think we need to bother with making distinction between not supported and unreachable for the libraries PAL.

Reply:
Yes, I went through all of these (had to remove `// NOTSUPPORTEDAPI` marker anyways) and all looks like using `errno = ENOTSUP` pattern.

The only one that needs a failfast is `void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)`